src/CartesianProduct.jl

```diff
@@ -562,3 +563,143 @@ function same_block_structure(x::AbstractVector{S1}, y::AbstractVector{S2}
     end
     return true
 end
+
+"""
+    block_structure(cpa::CartesianProductArray{N}) where {N}
+
+Returns an array containing the dimension ranges of each block in a `CartesianProductArray`.
+
+### Input
+
+- `cpa` -- Cartesian product array
+
+### Output
+
+A vector of ranges
+
+### Example
+
+```jldoctest
+julia> using LazySets: block_structure
+
+julia> cpa = CartesianProductArray([BallInf(zeros(n), 1.0) for n in [3, 1, 2]]);
+
+julia> block_structure(cpa)
+3-element Array{UnitRange{Int64},1}:
+ 1:3
+ 4:4
+ 5:6
+```
+"""
+function block_structure(cpa::CartesianProductArray{N}) where {N}
+    result = Vector{UnitRange{Int}}(undef, length(array(cpa)))
+    start_index = 1
+    @inbounds for (i, bi) in enumerate(array(cpa))
+        end_index = start_index + dim(bi) - 1
+        result[i] = start_index:end_index
+        start_index = end_index + 1
+    end
+    return result
+end
+
+
+"""
+    block_to_dimension_indices(cpa::CartesianProductArray{N}, vars::Vector{Int}) where {N}
+
+Returns a vector mapping block index `i`
+to tuple `(f, l)` such that either `f = l = -1` or `f` is the first dimension index and `l` is the last dimension
+index of the `i`-th block, depending on whether one of the block's dimension indices is specified in `vars`.
+
+### Input
+
+- `cpa` -- Cartesian product array
+- `vars` -- list containing the variables of interest, sorted in ascending order
+
+### Output
+
+A vector of tuples, where values in tuple relate to range of dimensions in the i-th block.
```

Comment:
The text doesn't correspond to the output (see line `630` below). The vector of tuples is only the first component of the output. Please define "non_empty_length" and add it in the docstring (or remove it if it's not needed).